src/parser/core/modules/AoEUsages.tsx

```diff
@@ -0,0 +1,139 @@
+	 */
+	stAbilities: Action[]
+	/**
+	 * The minimum number of targets the AoE ability should hit per cast.
+	 */
+	minTargets: number
+}
+
+/**
+ * This module checks for usages of AoE skills that have single target alternatives,
+ * such as BLMs Xenoglossy or NINs Hellfrog Medium, to ensure they hit the minimum
+ * number of targets.  It should not be used to check skills that can hit AoE but do
+ * not have alternatives, such as DRGs Dragonfire Dive or SMNs Deathflare, as it is
+ * correct to use those skills on single targets.
+ *
+ * For the AoE combos of melee DPS and Tanks, define a trackedAbility with the first
+ * abilities of the AoE and single target combos.  This is to provide leeway to finishing
+ * AoE combos when the number of targets may drop below the minimum during the combo.
+ */
+export abstract class AoEUsages extends Module {
+	static handle = 'aoeusages'
+	static title = t('core.aoeusages.title')`AoE Ability Usage`
+
+	@dependency private suggestions!: Suggestions
+
+	/**
+	 * Implementing modules MUST define the icon to be used for the suggestion.
+	 */
+	abstract suggestionIcon: string
+	/**
+	 * Implementing modules MUST define the abilities that are to be monitored.
+	 */
+	abstract trackedAbilities: AoeAbility[]
+
+	/**
+	 * Implementing modules MAY override the severity tiers for incorrect AoE usages.
+	 */
+	protected severity: SeverityTiers = {
+		1: SEVERITY.MINOR,
+		4: SEVERITY.MEDIUM,
+		7: SEVERITY.MAJOR,
+	}
+
+	private badUsages: Map<number, number> = new Map<number, number>()
+
+	protected init() {
+		this.addHook('aoedamage', {by: 'player', abilityId: this.trackedAbilities.map(a => a.aoeAbility.id)}, this.onAbility)
+		this.addHook('complete', this.onComplete)
+	}
+
+	/**
+	 * Implementing modules that have special logic where the number of minimum
+	 * targets for a skill is not constant may override this as needed.
+	 * @param event The event for which the number of minimum targets is being adjusted.
+	 * @param minTargets The default number of minimum targets for the ability as defined in trackedAbilities.
+	 */
+	protected adjustMinTargets(event: AoeEvent, minTargets: number) {
+		return minTargets
+	}
+
+	private onAbility(event: AoeEvent) {
+		const tracked = this.trackedAbilities.find(a => a.aoeAbility.id === event.ability.guid)
+
+		if (tracked === undefined) { return }
+
+		const minTargets = this.adjustMinTargets(event, tracked.minTargets)
+		if (event.successfulHit && event.hits.length < minTargets) {
+			this.badUsages.set(event.ability.guid, (this.badUsages.get(event.ability.guid) || 0) + 1)
+		}
+	}
+
+	private onComplete() {
+		const totalBadUsages = Array.from(this.badUsages.values()).reduce((acc, cur) => acc + cur, 0)
+
+		this.suggestions.add(new TieredSuggestion({
+			icon: this.suggestionIcon,
+			content: <Trans id="core.aoeusages.suggestion.content">
+				Use single target abilities unless there are enough targets for the AoE alternatives to do more damage.
```

Comment:
I wonder if this can be fleshed out more. Feels a bit terse, perhaps.

Reply:
I am open for alternate ideas.  I was aiming for the shortest thing that still says everything it needs to.  I don't think listing target counts here would make much sense, given that they can vary by skill or situation.

Reply:
I wanna say to avoid trying to be short in suggestion `content`. Will agree about listing target counts, though.

This is a total off-the-top-of-my-head attempt, but something like "Avoid using AoE actions when an alternative would do more damage. Many AoE actions share resources and/or cooldowns with a single-target action."

That's not a great example at all, but the point is to try and explain why the suggestion is a suggestion, as well as the path to fix it.

The `why` property itself is more "why this suggestion was triggered", rather than "why this suggestion is a suggestion".

Reply:
I'm still trying to come up with something for the content, but i wanted to clarify about the `why`. Are you making a general statement about `why` properties or saying that this suggestion does not currently have a good `why`?

Reply:
Just general while I was thinking about it

Reply:
how about:

> Avoid using AoE abilities when they would do less damage than an alternative single-target ability that shares a resource cost or cooldown.

for the suggestion content and changing the table description to:

> Some AoE abilities share resource costs or cooldowns with a single-target ability.  If the AoE skill cannot hit enough targets, the single-target ability will do more total damage and should be used instead.

Reply:
Better. I get the sense a bit of that detail could be moved from the table desc to the suggestion but whatever, same same.